rdrf/rdrf/forms/dynamic/fields.py

```diff
@@ -44,17 +45,23 @@ def validate(self, value):
 
 class FileTypeRestrictedFileField(FileField):
 
+    def _find_whitelisted_file_types(self, file_extension):
+        return WhitelistedFileExtension.objects.filter(file_extension__iexact=file_extension)
+
     def _find_blacklisted_mime_type(self, mt):
         from rdrf.models.definition.models import BlacklistedMimeType
         return BlacklistedMimeType.objects.filter(mime_type=mt).first()
 
     def validate(self, value):
         if not value:
             return super().validate(value)
-        __, ext = os.path.splitext(value._name)
+        __, ext = os.path.splitext(value.name)
         mime_type = magic.from_buffer(value.file.read(2048), mime=True)
         value.file.seek(0)
 
+        if not self._find_whitelisted_file_types(ext):
+            raise ValidationError(_(f"{ext} is not a supported file extension."))  # contact details if they believe it should be supported?
+
         blacklisted_mime_type = self._find_blacklisted_mime_type(mime_type)
```

Comment:
Do we want to get rid of the BlacklistedMimeType model and related logic?

Reply:
Yes, as per discussion

Reply:
A new setting to capture the CURATOR_EMAIL has been created to support this. The registry repos will need to be updated with relevant contacts for mnd, arrk, angelman.

Reply:
Blacklisted mime type logic has now been removed.